jobs/webapp-test.groovy

```diff
@@ -245,6 +245,8 @@ def runTestServer() {
          // tests.
          sh("deploy/trivial_diffs.py ${exec.shellEscape(params.BASE_REVISION)} ${exec.shellEscape(GIT_SHA1)} > ../trivial_diffs.txt");
          sh("git diff --name-only --diff-filter=ACMRTUB ${exec.shellEscape(params.BASE_REVISION)}...${exec.shellEscape(GIT_SHA1)} | fgrep -vx -f ../trivial_diffs.txt | testing/all_tests_for.py - > ../files_to_test.txt");
+         // Sometimes we need to run some extra tests for deletd files.
+         sh("git diff --name-only --diff-filter=D ${exec.shellEscape(params.BASE_REVISION)}...${exec.shellEscape(GIT_SHA1)} | fgrep -vx -f ../trivial_diffs.txt | testing/all_tests_for.py --deleted-mode - >> ../files_to_test.txt");
```

Comment:
Nit (or a question): Will the trivial_diffs.txt potentially contain deletions? if not, it seems unnecessary; if so, that would trim those out before running the command?

Reply:
It's a little hard to reason about that.  In theory, trivial_diffs.txt could contain a deleted file if it *only* consisted of sync-tags, with nothing between them.  But obviously that would never happen in practice.  If it did, it would be safe to filter it out since a file that consists solely of sync tags can't depend on another file.

I'd like to keep the code the way it is for symmetry, if nothing else.  It may be that the trivial-diff filtering is unnecessary, but I feel this block of code is easier to reason about, as a whole, if we keep lines 247 and 249 as similar as possible.

Reply:
Cool, thanks for the clarification! Good by me!